### DISCUSSION TITLE
Jailbreaking Juiceboxes?

### DISCUSSION THREAD

**Discussion by hefferbub** [citrineos/citrineos#11] [Q&A]
This project looks great, but what I don’t see is any discussion about how to get commercial chargers reconfigured to work with it.

We have about 30 Juicebox chargers and are looking to use a different backend with them, now that EnelX has destroyed their previously-excellent infrastructure.

Can you point me in the direction of folks who have experience with this?

Thanks!

**Comment by thanaParis** [comment]
Hi @hefferbub ! Thank you for this question.
CitrineOS is a OCPP 2.0.1 CSMS, designed to work with any chargers that implement OCPP 2.0.1. There are many 2.0.1 commercial chargers out there, but the majority are OCPP 1.6 only, which CitrineOS does not support.

Looking at the Juicebox website, their chargers appear to only support OCPP 1.6.

It sucks that EnelX has taken down their backend. This reinforces everything that went into the decision to make CitrineOS: it is essential to have open source solutions available for when private companies cease to exist or support their hardware. These open source solutions however depend on the use of open standards such as OCPP when making chargers. OCPP 1.6 lacks support for many important features, such as device monitoring and ISO15118 'Plug and Charge'. While Juicebox made their chargers OCPP 1.6 compatible most likely they customized and extended the protocol, which would mean not all features their backend offered can be supported through 1.6 alone. If they were OCPP 2.0.1 compatible, you would be able to control every aspect of the charger no matter what.

[StEVe](https://github.com/steve-community/steve) is a OCPP 1.6 open source CSMS that should allow you to at least make basic use of your chargers. If that turns out to be insufficient, I would take a look at the [EVerest](https://github.com/EVerest/everest) community. EVerest is a open source charging station firmware project, and it may be possible to jailbreak your Juicebox chargers to use it instead of the software they are shipped with. The EVerest community is very active, so although I don't think anyone has tried that before if you post asking how to do it I believe you will get very good feedback and support. If you do end up with EVerest running, you can use their OCPP 2.0.1 layer and then you will be able to leverage CitrineOS.
